client.go

```diff
@@ -72,13 +81,15 @@ type client struct {
 	seedBrokers []*Broker
 	deadSeeds   []*Broker
 
-	brokers  map[int32]*Broker                       // maps broker ids to brokers
-	metadata map[string]map[int32]*PartitionMetadata // maps topics to partition ids to metadata
+	brokers      map[int32]*Broker                       // maps broker ids to brokers
+	metadata     map[string]map[int32]*PartitionMetadata // maps topics to partition ids to metadata
+	coordinators map[string]*Broker                      // Maps consumer group names to coordinating brokers
```

Comment:
If the coordinator call returns a new broker, it is always registered in the brokers array as well. What would be the benefit of just storing the ID? AFAICS it's a tradeoff between an extra hash lookup vs. a bit more memory.